addons/mass_mailing/models/mail_notification.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+
+
+class MailNotification(models.Model):
+    # Extend the mail.notification model for tracking (marketing purpose)
+    _inherit = 'mail.notification'
+
+    # mail data
+    mail_id_int = fields.Integer(
+        string='Mail ID (tech)',
+        help='ID of the related mail_mail. This field is an integer field because '
+             'the related mail_mail can be deleted separately from its statistics. '
+             'However the ID is needed for several action and controllers.',
```

Comment:
This is only necessary to keep "traces" right?
Can you give some examples of where it would be needed in action and controllers? I could not find any :(

Also, we usually use triple quotes for multi-line help attributes (small detail).

Reply:
It used in the _get_records method, used by trace controller indirectly (set_opened, ...).

Also for the triple quotes, I prefer to not change because it adds lot of space characters in the help string (it is trim and clean after but I find that ugly).